pkg/blob/fetch.go

```diff
@@ -65,7 +73,18 @@ func (f *Fetcher) Fetch(dir string, blobURL string) error {
 		if err != nil {
 			return err
 		}
+
+		if !isTar(r){
```

Comment:
i like keeping `isTar` outside of `extractTar` because
- `isTar` requires a `Seek(0,0)` to reset the reader but if we use a `ReadSeeker` as the arg to `extractTar` we lose the ability to use `gzip.NewReader(r)` as the reader which is nice to have.
- I think `isTar` is meant as a last check validation that the file is the type we expect, which I think is a different responsibility than validating that the input to `extractTar` is a tar file. we're able to cleanly handle the `unexpectedBlobTypeError` in this way.

Reply:
@matthewmcnew

Reply:
That makes sense.